control-plane/agents/src/bin/core/volume/specs.rs

```diff
@@ -828,6 +829,31 @@ impl ResourceSpecsLocked {
         }
     }
 
+    pub(crate) fn check_capacity_limit_for_resize(
+        &self,
+        request: &ResizeVolume,
+        current_borrowed_limit: u64,
+    ) -> Result<(), SvcError> {
+        // We anyway called this function after checking cluster_capacity_limit is not None,
+        // but let's keep this match for completion, should the caller end change at some point.
+        match request.cluster_capacity_limit {
```

Comment:
We don't need request and do match again, just use the limit, see my suggestion above.